server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/ScheduledTaskLoggerService.java

```diff
@@ -213,15 +235,65 @@ private File getLogFile(Long scheduleTaskId, OdcTaskLogLevel level) {
         ExecutorInfo executorInfo = JsonUtils.fromJson(taskEntity.getExecutor(), ExecutorInfo.class);
         if (!dispatchChecker.isThisMachine(executorInfo)) {
             try {
-                DispatchResponse response =
-                        requestDispatcher.forward(executorInfo.getHost(), executorInfo.getPort());
-                return response.getContentByType(
-                        new TypeReference<SuccessResponse<File>>() {}).getData();
+                return forwardToGetLogContent(executorInfo.getHost(), executorInfo.getPort());
+            } catch (Exception e) {
+                log.warn("Remote get task log failed, scheduleTaskId={}", scheduleTaskId, e);
+                throw new UnexpectedException(
+                        String.format("Remote interrupt task failed, scheduleTaskId=%s", scheduleTaskId));
+            }
+        }
+        File logFile = getLogFileFromCurrentMachine(taskEntity, level);
+        return LogUtils.getLatestLogContent(logFile, loggerProperty.getMaxLines(), loggerProperty.getMaxSize());
+    }
+
+    private InputStream downloadLogFile(Long scheduleTaskId, OdcTaskLogLevel level) {
+        ScheduleTaskEntity taskEntity = scheduleTaskService.nullSafeGetById(scheduleTaskId);
+        if (taskFrameworkEnabledProperties.isEnabled() && taskEntity.getJobId() != null) {
+            try {
+                final InputStream[] logStreams = new InputStream[1];
```

Comment:
why need array here?

Reply:
this is a lambda function, if I want to get inner value, I must set a final variable